packages/devtools-evm-hardhat/src/tasks/deploy.ts

```diff
@@ -177,11 +180,35 @@ const action: ActionType<TaskArgs> = async (
                 // We need to make sure the user has enabled hardhat-deploy
                 assertHardhatDeploy(env)
 
+                // We first collect all existing deployments
+                //
+                // We do this so that we can diff the state before and after
+                // running the deployment scripts.
+                //
+                // This is, in immediate effect, a workaround for having to set resetMemory
+                // in the options for the run() function below to false. In near future though
+                // it opens doors for being able to return partially successful deployment results
+                const deploymentsBefore = await env.deployments.all()
+
                 // The core of this task, running the hardhat deploy scripts
-                const contracts = await env.deployments.run(selectedTags, {
+                const deploymentsAfter = await env.deployments.run(selectedTags, {
+                    // If we don't pass resetmemory or set it to true,
+                    // hardhat deploy will erase the database of deployments
+                    // (including the external deployments)
+                    //
+                    // In effect this means the deployments for LayerZero artifacts would not be available
+                    resetMemory: false,
                     writeDeploymentsToFiles: true,
+                    deletePreviousDeployments: reset,
                 })
 
+                // Now we do a little diff on what contracts had been changed
+                const contracts = Object.fromEntries(
+                    Object.entries(deploymentsAfter).filter(
```

Comment:
This does not take into account if `Object.entries(deploymentsAfter)` is a subset of `deploymentsBefore`.  I.e., if `deploymentsBefore` actually has more deployments than after.  Thus, it is an incomplete diff.

Reply:
Yeah this logic is very basic and it only cares about the new and/or changed deployments - we are not telling the user which contracts have been deleted.

Reply:
Got it, so long as you are aware of and fine with this limitation, LGTM